### PR TITLE
UCS: added ucs_basename function

### PR DESCRIPTION
**Pull request by hoopoepg**
- added ucs_basename function which may operate by const char string

### CODE REVIEW

File: src/ucm/util/log.c [hunk 1]
```diff
@@ -258,7 +259,7 @@ void __ucm_log(const char *file, unsigned line, const char *function,
     gettimeofday(&tv, NULL);
     ucm_log_snprintf(buf, UCM_LOG_BUG_SIZE - 1, "[%lu.%06lu] [%s:%d] %18s:%-4d UCX  %s ",
                      tv.tv_sec, tv.tv_usec, ucm_log_hostname, getpid(),
-                     basename(file), line, ucm_log_level_names[level]);
+                     ucs_basename(file), line, ucm_log_level_names[level]);
```
Comment (hoopoepg) [inline src/ucm/util/log.c]: here was issue: basename operates by mutable string, really don't know how it compiled before

File: src/ucs/sys/string.h [hunk 2]
```diff
@@ -128,6 +129,22 @@ char* ucs_strncpy_safe(char *dst, const char *src, size_t len);
...
+ * @param path Path to parse.
+ * 
+  * @return file name
+*/
+static UCS_F_ALWAYS_INLINE const char* ucs_basename(const char *path)
+{
+    const char *name = strrchr(path, '/');
+
+    return (name == NULL) ? path : name + 1;
+}
```
Comment (kostikbel) [inline src/ucs/sys/string.h]: This is very naive implementation of basename(3).  I cannot say if it is enough for the purposes of UCX, or do the callers depend on the full POSIX semantic.  But I can point out that, for example,  POSIX requires that NULL or empty string argument returns ".", that "/" and "//" returned "/", and that trailing '/' characters were deleted.

Comment (hoopoepg) [reply src/ucs/sys/string.h]: this is replacement enough for internal use only, we had a lot of same implementations of this function, I just merged it into single location

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11538/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 02-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/165977b8d985cf9509a93b9624ee5d7f9e26a493/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/16ef45b98f26ac3c1189973b819f13b51473d7bb/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/2651f1104986db36d4e5048a392604f043396e24/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/7fc86c344ffa28c6e6fa937257dcad30d0459366/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/f719bb6e5b9c27d19031a3f775a1dfa16db754ea/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/3fb42f2871bc3521f133540946c01df7952e4f20/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/2729d7ed6241f9321ac41f2a79b05e16f2f03e2d/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/d6f5808fa7ae7e21995254993dc68d4810f4dbc9/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/8cee8239f1a0f321742415b77ae690d0dac88528/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/58f0018f193d1b155d727e1c42d24e552adcd548/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/b0348b0506e297d611bba461700a28d5484209c8/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/ecd97e2846768a98587608b3ebf4fc59943dae65/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/27d9b4b55f157c48cc1fd3ae6fd32bbd02b9df3c/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/3ec51f223efac68692c42b02a497bd0dc1316f84/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/c67cf270d0e6b95fc2fe13768000dbf042aa4e40/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/bbd68bdfd0a75643b0f155469c25417119df4473/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/2ef045b33d5d229bca26925c8ef64d26d1d225ef/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/3c3fc1d71ec346cab3d284e2ecfeab6b645d6fff/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/06d02114525ff5e70fa64f1babcdffebfc78d055/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/23018146d5c48969de1150d7bda33dbabfd58c44/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/425da474af630691bf96b4a6d9cd3a0a4692f392/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/bdb47e8f8b9d72f7395e759bb01e2fbcd7d81147/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/5295e8a17ca3df8260d42cb6f3d1dab3afcb821a/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/b313cf3846e6d6eff7f1a0a33a40c44ad685d105/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/b50724b06db6986ee8ed3fcc0ccd3dc9f5ed20d5/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/52125589c4b3c283e1c1553a3a27b03640e7db7b/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/752a3c8cf1a5a724f80d39626e9093371fb8226f/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/dbe122a6fd675a05fd08f0157164114a63107e26/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/16b34bdbd93b45fa0650f6ce6d0c3340/raw/1da48a38ac06e82baa7720c8b163797a7d1d3042/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11540/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 02-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/5f5773bf251a5812bd989a5462bd2ff943dc3994/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/5cf2d2d683c495547a544d27f13003808535a348/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/622d767efbfd1820ebefce929bd063a313906065/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/cf13175ee038fdb5f90fc3a1e877369a086691fe/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/99b0f8e055608e07d204d20ace9af69d4076e903/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/40ae0f0f5dae2a25305da91b829f588555912919/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/de4741db5910aba8d1c749950fa8da3bc91174a4/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/4abc1ce00e3764cdbd4a004b47b3cf8667bc275f/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/b5645a294a0a217b322cb28541d75be26f487a34/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/951e0eb06673cf3285c1dec7f7d2e0a8f4db2876/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/9f432f2e92357b660e79a83bd03aab484d3b2558/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/233e2d12266ff18f553a189bd6cb7ac86f5b4c54/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/6aa351041704ef2e181f67150abe6db208816a0b/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/588179ffb1c27e957f4583ed8c04c09f8b3fb99c/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/e7f1e8554924d19de9e2636a81d03ec24b2ffa83/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/fad7b6b44ee77abd83b835b4c34f9ac02927355e/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/31205279c62e32066a6a40194d72c0b81e5bebfa/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/7ea2d5ceb9c92d502aa8bafd8fbbb61c0255e00a/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/90f2c1bf64bc05a30ead24a5b3ab6b5616a8f868/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/91cb0baa30dd4e3998d2a57fd13c3886cdab9dc9/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/0ae7e8f2575b39775e8f137ceb37a1f1653824a1/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/cb9da3eb95fc03b27ad4c489f1fe11ec5518336a/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/4fa716c129283d9ee0c1183858018559adea293b/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/1f1298b215fb9fb9c67b5202c3f47caf1ab2195c/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/0c0fd99b3355f6e781cff04abf98d43cf899331c/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/6a355661e909b07359f8853fba36969c6ae26aac/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/e13c1fde0956c6cffcc74015b836e85453883980/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/9ac8ee96b0d242c79b0e3f3ca70f392e74fc4408/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/d11240022a9368b6258702b10d8b5d7e/raw/a3271650fa098d74a0302b16c55b7ac27c3e8366/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**